graphql/util.go

```diff
@@ -1310,6 +1301,34 @@ func hasAnnotationPermission(ctx context.Context, obj *restModel.APITask, requir
 	return false, nil
 }
 
+func hasLogViewPermission(ctx context.Context, obj *restModel.APITask) (bool, error) {
+	authUser := gimlet.GetUser(ctx)
+	permissions := gimlet.PermissionOpts{
+		Resource:      *obj.ProjectId,
+		ResourceType:  evergreen.ProjectResourceType,
+		Permission:    evergreen.PermissionLogs,
+		RequiredLevel: evergreen.LogsView.Value,
+	}
+	if authUser.HasPermission(permissions) {
+		return true, nil
+	}
+	return isPatchAuthorForTask(ctx, obj)
```

Comment:
Is the implication here that patch authors can always view their own logs even if they don't have permission to view other logs in the project? This seems like a security loophole.

Reply:
That's right. This is what we do for task annotations. The user needs the [Patch Submit](https://github.com/evergreen-ci/evergreen/blob/f9f7299af25bd7a65750c0fd20a0c8b6bc37ff40/globals.go#L1258) permission to submit a task. Sometimes we grant bots the ability to submit patches, maybe this is a potential loophole.